util/src/main/java/org/hps/util/CrystalBallFunction.java

```diff
@@ -0,0 +1,269 @@
+package org.hps.util;
+
+import hep.aida.IAnalysisFactory;
+import hep.aida.IAnnotation;
+import hep.aida.IFitFactory;
+import hep.aida.IFitResult;
+import hep.aida.IFitter;
+import hep.aida.IFunction;
+import hep.aida.IFunctionFactory;
+import hep.aida.IHistogram1D;
+import hep.aida.IHistogramFactory;
+import hep.aida.IPlotter;
+import hep.aida.IPlotterStyle;
+import hep.aida.ITree;
+import hep.aida.ITreeFactory;
+import hep.aida.ref.Annotation;
+import static java.lang.Math.PI;
+import static java.lang.Math.abs;
+import static java.lang.Math.exp;
+import static java.lang.Math.pow;
+import static java.lang.Math.sqrt;
+import java.util.Random;
+import static org.apache.commons.math3.special.Erf.erf;
+
+/**
+ *
+ * @author Norman Graf
+ */
+public class CrystalBallFunction implements IFunction, Cloneable {
+
+    double xmean;
+    double sigma;
+    double n;
+    double alpha;
+    double Norm;
+
+    String[] vars = {"xmean", "sigma", "n", "alpha", "Norm"};
+
+    @Override
+    public String title() {
+        return "Crystal Ball Function";
+    }
+
+    @Override
+    public void setTitle(String string) throws IllegalArgumentException {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public double value(double[] x) {
+
+        double s = (x[0] - xmean) / sigma;
+        double A = pow((n / abs(alpha)), n) * exp(-0.5 * abs(alpha) * abs(alpha));
```

Comment:
Might be worth checking if recursive power function is faster. Might be beneficial when doing multiple fits.